include/rdma/fabric.h

```diff
@@ -449,6 +456,7 @@ struct fi_domain_attr {
 	size_t			max_err_data;
 	size_t			mr_cnt;
 	uint32_t		tclass;
+	uint32_t		max_group_id;
```

Comment:
This change breaks ABI compatibility and requires an updated abi_compat struct.  We can limit the break by coordinating these changes with #9319, which also updates the structure, in order to avoid breaking the ABI twice.